src/platform/stm32f1/rcc.c

```diff
@@ -80,5 +83,138 @@ void rcc_init(enum stm32f1_external_clock_value clock)
+			u32 pllsrc = RCC->CFGR & RCC_CFGR_PLLSRC;
+
+			if (pllsrc == 0) /* 0 means PLL INPUT = HSI / 2 */
+			{
+				rcc_clock_tree.sys_clock_freq = (HSI_VALUE >> 1) * pllmul;
+			} else /* 1 means PLL INPUT = HSE, HSE divided by 2 if XTPRE = 1 */
+			{
+				if ((RCC->CFGR & RCC_CFGR_PLLXTPRE) == RCC_CFGR_PLLXTPRE_HSE_DIV2)
+					rcc_clock_tree.sys_clock_freq = (EXTERNAL_CLOCK_VALUE >> 1) * pllmul;
+				else
+					rcc_clock_tree.sys_clock_freq = EXTERNAL_CLOCK_VALUE * pllmul;
+			}
+			break;
+		}
+	}
+
+	/* Determine ahb bus clock */
+	switch (RCC->CFGR & RCC_CFGR_HPRE) {
+		case RCC_CFGR_HPRE_DIV1:
+			rcc_clock_tree.ahb_clock_freq = rcc_clock_tree.sys_clock_freq;
+			break;
+		case RCC_CFGR_HPRE_DIV2:
+			rcc_clock_tree.ahb_clock_freq = rcc_clock_tree.sys_clock_freq >> 1;
+			break;
+		case RCC_CFGR_HPRE_DIV4:
+			rcc_clock_tree.ahb_clock_freq = rcc_clock_tree.sys_clock_freq >> 2;
+			break;
+		case RCC_CFGR_HPRE_DIV8:
+			rcc_clock_tree.ahb_clock_freq = rcc_clock_tree.sys_clock_freq >> 3;
+			break;
+		case RCC_CFGR_HPRE_DIV16:
+			rcc_clock_tree.ahb_clock_freq = rcc_clock_tree.sys_clock_freq >> 4;
+			break;
+		case RCC_CFGR_HPRE_DIV64:
+			rcc_clock_tree.ahb_clock_freq = rcc_clock_tree.sys_clock_freq >> 6;
+			break;
+		case RCC_CFGR_HPRE_DIV128:
+			rcc_clock_tree.ahb_clock_freq = rcc_clock_tree.sys_clock_freq >> 7;
+			break;
+		case RCC_CFGR_HPRE_DIV256:
+			rcc_clock_tree.ahb_clock_freq = rcc_clock_tree.sys_clock_freq >> 8;
+			break;
+		case RCC_CFGR_HPRE_DIV512:
+			rcc_clock_tree.ahb_clock_freq = rcc_clock_tree.sys_clock_freq >> 9;
+			break;
+			gcc global static variable
+	}
+
+	/* Determine apb bus 1 clock */
+	switch (RCC->CFGR & RCC_CFGR_PPRE1) {
+		case RCC_CFGR_PPRE1_DIV1:
+			rcc_clock_tree.apb1_clock_freq = rcc_clock_tree.ahb_clock_freq;
+			break;
+		case RCC_CFGR_PPRE1_DIV2:
+			rcc_clock_tree.apb1_clock_freq = rcc_clock_tree.ahb_clock_freq >> 1;
+			break;
+		case RCC_CFGR_PPRE1_DIV4:
+			rcc_clock_tree.apb1_clock_freq = rcc_clock_tree.ahb_clock_freq >> 2;
+			break;
+		case RCC_CFGR_PPRE1_DIV8:
+			rcc_clock_tree.apb1_clock_freq = rcc_clock_tree.ahb_clock_freq >> 3;
+			break;
+		case RCC_CFGR_PPRE1_DIV16:
+			rcc_clock_tree.apb1_clock_freq = rcc_clock_tree.ahb_clock_freq >> 4;
+			break;
+			gcc global static variable
+	}
+
+	/* Determine apb bus 2 clock */
+	switch (RCC->CFGR & RCC_CFGR_PPRE2) {
+		case RCC_CFGR_PPRE2_DIV1:
+			rcc_clock_tree.apb2_clock_freq = rcc_clock_tree.ahb_clock_freq;
+			break;
+		case RCC_CFGR_PPRE2_DIV2:
+			rcc_clock_tree.apb2_clock_freq = rcc_clock_tree.ahb_clock_freq >> 1;
+			break;
+		case RCC_CFGR_PPRE2_DIV4:
+			rcc_clock_tree.apb2_clock_freq = rcc_clock_tree.ahb_clock_freq >> 2;
+			break;
+		case RCC_CFGR_PPRE2_DIV8:
+			rcc_clock_tree.apb2_clock_freq = rcc_clock_tree.ahb_clock_freq >> 3;
+			break;
+		case RCC_CFGR_PPRE2_DIV16:
+			rcc_clock_tree.agcc global static variablepb2_clock_freq = rcc_clock_tree.ahb_clock_freq >> 4;
+			break;
+	}
+
+	/* Determine apb bus 1 tim clock */
```

Comment:
nitpick: Can we use the peripheral names here to make things easier to identify? In this case, TIM1.